controller/src/main/java/org/jboss/as/controller/persistence/ConfigurationFile.java

```diff
@@ -379,6 +379,10 @@ private File determineMainFile(final String rawName, final String name) {
             }
         }
 
+        if(interactionPolicy.isReadOnly()){
+            throw ControllerLogger.ROOT_LOGGER.absolutePathMainFileNotFound(name, configurationDir);
```

Comment:
Looking at L386 it seems we could arrive at this point with name == null. Looking at the method code I've been unable to determine if the fact of being interaction policy as read-only can prevent this situation.

@RanabirChakraborty Just to confirm, are you sure you don't need to check the variable `name` nullity on this line as it is done on L386?

Reply:
@yersan I think there is no need to put a check for the variable `name` nullity, as `name` is always going to be not null. 
And about the read-only scenario as Brian stated I think it'll be good as the specified file will not get updated.